solr/core/src/test/org/apache/solr/search/function/distance/NVectorDistTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.search.function.distance;
+
+import org.apache.solr.SolrTestCaseJ4;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Locale;
+
+public class NVectorDistTest extends SolrTestCaseJ4 {
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    System.setProperty("enable.update.log", "false"); // schema12 doesn't support _version_
+    initCore("solrconfig-nvector.xml", "schema-nvector.xml");
+  }
+
+  @Override
+  public void setUp() throws Exception {
+    Locale.setDefault(Locale.ENGLISH);//for parsing lat/log correctly in this test
+    super.setUp();
+    assertU(delQ("*:*"));
+    assertU(commit());
+  }
+
+  @Test
+  public void testNVector() throws Exception {
+    assertU(adoc("id", "0", "nvector", "52.02471051274793, -0.49007556238612354"));
+    assertU(commit());
+    assertJQ(
+        req("defType", "lucene", "q", "*:*", "fl", "id,nvector*", "sort", "id asc"),
+        "/response/docs/[0]== {"
+            + "'id':'0',"
+            + "'nvector_0_d1':0.6152990562577377,"
+            + "'nvector_1_d1':-0.005263047078845837,"
+            + "'nvector_2_d1':0.7882762026750415,"
+            + "'nvector':'52.02471051274793, -0.49007556238612354'}");
+
+    assertJQ(
+        req(
+            "defType", "lucene",
+            "q", "*:*",
+            "nvd", "nvdist(52.01966071979866, -0.4983083573742952,nvector)",
```

Comment:
I missed where 'nvdist' is defined as a function. Do we need to add to `ValueSourceParser` static init block?

Reply:
added to ValueSourceParser